rust-connectors/common/src/opt.rs

```diff
@@ -288,3 +305,27 @@ impl CommonSourceOpt {
         Ok(consumer.stream_with_config(offset, config).await?)
     }
 }
+use schemars::schema_for;
+
```

Comment:
@morenol What do you think of this trait? It'd enable a pattern to show both metadata and options [for a given connector options struct](https://github.com/infinyon/fluvio-connectors/pull/278/files#diff-39410f653ec587700f2cc9571db2616ad92b78f29bda51334c3f9dee18641c09R51-R61) and [in the connector's main](https://github.com/infinyon/fluvio-connectors/pull/278/files#diff-70914ecb8d394bfb11424c6971491c907394261f97f31656a36fc4959e9a98b6R16-R20).

Reply:
I should probably do this in a different PR but I've bene thinking about doing this for a while and wanted to see what it looked like given how many changes are going into the common crate.